### PR TITLE
Add setting to enable swipe to go back in Look and Feel settings

### PR DESCRIPTION
**Pull request by sthomas727**
Implemented the functionality that I requested in Issue #1117 , and was also discussed in #979 .

### CODE REVIEW

File: app/schemas/com.jerboa.db.AppDB/25.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "formatVersion": 1,
   "database": {
     "version": 25,
-    "identityHash": "e7e957b63c2297483c5e9aeb1e263a1f",
```
Comment (MV-GH) [inline app/schemas/com.jerboa.db.AppDB/25.json:5]: Increase the DB version, do not change the existing scheme

File: app/src/main/java/com/jerboa/MainActivity.kt [hunk 2]
```diff
@@ -493,7 +493,11 @@ class MainActivity : AppCompatActivity() {
                         ),
                     ) {
                         val args = Route.PostArgs(it)
-                        SwipeToNavigateBack(appState::navigateUp) {
+
+                        SwipeToNavigateBack(
+                            appSettings.useSwipeBack,
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/MainActivity.kt:498]: Make it a enum, later when full swipe gestures gets added we can just add it to this enum.

File: app/src/main/java/com/jerboa/db/entity/AppSettings.kt [hunk 3]
```diff
@@ -113,4 +113,9 @@ data class AppSettings(
         defaultValue = "0",
     )
     val autoPlayGifs: Boolean,
+    @ColumnInfo(
+        name = "use_swipe_back",
+        defaultValue = "1",
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/db/entity/AppSettings.kt:118]: It should default to no gestures

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -203,6 +203,7 @@
     <string name="look_and_feel_theme_color_woodland">Woodland</string>
     <string name="look_and_feel_use_custom_tabs">Use custom tabs</string>
     <string name="look_and_feel_use_private_tabs">Use private custom tabs if available</string>
+    <string name="look_and_feel_use_swipe_back">Use swipe to navigate back on posts</string>
```
Comment (MV-GH) [inline app/src/main/res/values/strings.xml:206]: Name it something like Post gestures, and then have `none option` and `swipe back option`

**Review by MV-GH — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (MV-GH) [reply app/src/main/res/values/strings.xml:206]: You can find some ideas for the nomenclature here https://github.com/dessalines/jerboa/pull/872/files

File: app/schemas/com.jerboa.db.AppDB/25.json (hunk 1 above)
Comment (MV-GH) [reply app/schemas/com.jerboa.db.AppDB/25.json:5]: Also add migrations and test them

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (sthomas727) [reply app/src/main/res/values/strings.xml]: I've implemented it as you suggested, I've titled the list `Post Navigation Gestures`, and provided a `Disabled` option and a `Swipe Left to Navigate Back` option.

File: app/schemas/com.jerboa.db.AppDB/25.json (hunk 1 above)
Comment (sthomas727) [reply app/schemas/com.jerboa.db.AppDB/25.json]: You should see this reflected in my latest commit. The migration test passed.

File: app/src/main/java/com/jerboa/Utils.kt [hunk 5]
```diff
@@ -835,6 +835,18 @@ enum class PostViewMode(val mode: Int) {
     List(R.string.look_and_feel_post_view_list),
 }
 
+enum class NavigationGestureMode(val mode: Int) {
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/Utils.kt:838]: Rename to `PostNavigationGestureMode`

File: app/src/main/java/com/jerboa/ui/components/common/SwipeToNavigateBack.kt [hunk 6]
```diff
@@ -16,34 +16,41 @@ import androidx.compose.ui.Modifier
...
-            when (it) {
-                DismissValue.DismissedToEnd -> {
-                    onSwipeBack()
-                    true
-                }
-                else -> { false }
-            }
-        },
-    )
+    if (useSwipeBack == 1) {
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/common/SwipeToNavigateBack.kt:23]: SwipeLeft.ordinal

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -174,6 +174,9 @@
     <string name="look_and_feel_activity_show_content_for_collapsed_comments">Show content for collapsed comments</string>
     <string name="look_and_feel_font_size">Font size: %1$s</string>
     <string name="look_and_feel_look_and_feel">Look and feel</string>
+    <string name="look_and_feel_post_navigation_gesture_mode">Post Navigation Gestures</string>
+    <string name="look_and_feel_post_navigation_gesture_mode_disabled">Disabled</string>
+    <string name="look_and_feel_post_navigation_gesture_mode_swipe_left">Swipe Left to Navigate Back</string>
```
Comment (MV-GH) [inline app/src/main/res/values/strings.xml:179]: Only capitalize the first letter

**Review by MV-GH — CHANGES_REQUESTED** [review]
Few small changes and its ready

File: app/src/main/java/com/jerboa/Utils.kt (hunk 5 above)
Comment (MV-GH) [reply app/src/main/java/com/jerboa/Utils.kt:838]: And move it to its own file in /feat package

**Review by MV-GH — APPROVED** [review]

**Pull request merged**